### PR TITLE
added db config and github action ci

### LINKED ISSUE #2: create helm repo for unleash
I was not able to find unleash via https://artifacthub.io so it seems there is no helm repo available, which hosts the chart?

Did i just missed it?
Unfortunatly there is no Readme.md provided for the chart too, which contains install instructions.

If there is nor helm repo available jet, you could host the Helm chart via Github releases and Github pages.

I'm experienced with this from other repos (kokuwa, kiwigrid, prometheus-community and so on) so if wanted i coud provide an pr, implementing this via Github actions.

### PR DESCRIPTION
**Pull request by monotek**
Signed-off-by: André Bauer <monotek23@gmail.com>

* added dbconfig to be able to use external database
* added possibility to use exisitng db secrets
* changed label handling
* cleanup values file (removed superfluous postgres config & commented out stuff)
* added chart readme
* enabled readiness and livenessprobes
* added github action ci (codespell, superlinter, kubeval & helm chart testing)
* added github release to helm repo (needs gh-pages branch)
* added gitignore file
* added hpa
* disabled pod antiaffinity as only replica 1 is used per default
* changed chart foldername

fixes #2

how th pipeline works can be seen here: https://github.com/monotek/unleash-helm-charts/actions/runs/572469119

### CODE REVIEW

**Comment by evanstucker-hates-2fa** [comment]
This seems reasonable to me. It would be nice to get on artifacthub.io as an official, verified publisher.

**Comment by evanstucker-hates-2fa** [comment]
I'm talking to @ivarconr on unleash-community.slack.com about merging this.

File: charts/unleash/templates/deployment.yaml [hunk 1]
```diff
@@ -4,40 +4,40 @@ kind: Deployment
...
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Release.Name }}-postgresql
-                  key: postgresql-password
+                  name: {{ if .Values.dbConfig.useExistingSecret.name }}{{ .Values.dbConfig.useExistingSecret.name }}{{ else }}{{ .Values.postgresql.fullnameOverride }}{{ end }}
+                  key: {{ if .Values.dbConfig.useExistingSecret.key }}{{ .Values.dbConfig.useExistingSecret.key }}{{ else }}postgresql-password{{ end }}
+            {{- end }}                  
             - name: DATABASE_URL
-              value: postgres://postgres:$(POSTGRESQL_PASSWORD)@{{ .Release.Name }}-postgresql/unleash
+              value: postgres://{{ .Values.dbConfig.user }}:{{ if or (.Values.dbConfig.useExistingSecret.name) (.Values.postgresql.enabled) }}$(POSTGRESQL_PASSWORD){{ else }}{{ .Values.dbConfig.pass }}{{ end }}@{{ if .Values.postgresql.enabled }}{{ .Values.postgresql.fullnameOverride }}{{ else }}{{ .Values.dbConfig.host }}{{ end }}:{{ .Values.dbConfig.port }}/{{ .Values.dbConfig.database }}
```
Comment (evanstucker-hates-2fa) [inline charts/unleash/templates/deployment.yaml:40]: This line is pretty long. Perhaps we should switch to using DB_USER, DB_PASS, DB_HOSTNAME, DB_NAME and then build DATABASE_URL (if that's what Unleash is expecting) in the template.yaml?

**Review by evanstucker-hates-2fa — APPROVED** [review]

File: charts/unleash/templates/deployment.yaml (hunk 1 above)
Comment (evanstucker-hates-2fa) [reply charts/unleash/templates/deployment.yaml:40]: Not blocking the PR for this - it's just a comment.

Comment (monotek) [reply charts/unleash/templates/deployment.yaml:40]: I'll try to change it tomorrow :)

**Review by FredrikOseberg — APPROVED** [review]
LGTM. Need to make sure we enable gh-pages, otherwise one of the steps in the release action will fail.

**Comment by monotek** [comment]
Not sure but maybe this affected the docs on as https://unleash.github.io they are not available at the moment.

Maybe because the doamain is used twice now? 

As far is i understand unleash.github.io is already used in: https://github.com/Unleash/unleash.github.io

**Comment by ivarconr** [comment]
> Not sure but maybe this affected the docs on unleash.github.io as they are not available at the moment.

This was a bug as I was not aware CNAME feature in github required a file in repo, which was auto-generated on build. Should work again!

**Review by ivarconr — APPROVED** [review]

**Pull request merged**